src/base/os.lua

```diff
@@ -80,6 +80,9 @@
 			if os.is("macosx") then
 				formats = { "lib%s.dylib", "%s.dylib" }
 				path = os.getenv("DYLD_LIBRARY_PATH") or ""
+			elseif os.is("cygwin") then
+				formats = { "lib%s.a", "%s.a", "lib%s.dll.a", "%s.dll.a" }
+				path = os.getenv("LD_LIBRARY_PATH") or ""

```

Comment:
I'm not sure I'm sold on this. Why do you think os.is() shouldn't return 'windows', when it's running on windows?
Cygwin isn't an OS, it's a toolchain, but I guess it's a bit awkward that it also implies a few tools that make it feel more linux-like.